src/internal/repository-metadata-files/module/list.go

```diff
@@ -0,0 +1,41 @@
+package module
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"registry-stable/internal/module"
+)
+
+func ListModules() ([]module.Module, error) {
+	// TODO: Accept a directory as a parameter (or pass it in some other way) so we can easily test this
+	moduleDirectory := "../modules"
+
+	// walk the module directory recursively and find all json files
+	// for each json file, parse it into a module.Module struct
+	// return a slice of module.Module structs
+
+	var results []module.Module
+	err := filepath.Walk(moduleDirectory, func(path string, info os.FileInfo, err error) error {
+		if filepath.Ext(path) == ".json" {
+			// extract the module details from the file path. It should be modules/<firstLetterOfNamespace>/<namespace>/<name>/<system>.json
+			regex := regexp.MustCompile(`modules\/([a-z])\/([a-z0-9-]+)\/([a-z0-9-]+)\/([a-z0-9-]+).json`)
```

Comment:
I believe there can be capital letters in namespaces, no? I'm not sure.

Reply:
I believe they were in Arel's PoC, but maybe the github api is case-insensitive.

Reply:
Good catch, Ill update this regex :)

Reply:
However, maybe we should actually only store lowercase in the repository?

Reply:
I don't think there are uppercase namespaces in the hashi registry?

Reply:
Modules with capital letters in the namespace should be supported
Also, `_` should be supported